app/src/main/java/com/isc/hermes/OfflineModeSettingsActivityView.java

```diff
@@ -0,0 +1,275 @@
+package com.isc.hermes;
+
+import static com.isc.hermes.ActivitySelectRegion.MAP_CENTER_LATITUDE;
+import static com.isc.hermes.ActivitySelectRegion.MAP_CENTER_LONGITUDE;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.cardview.widget.CardView;
+
+import com.isc.hermes.model.RegionData;
+import com.isc.hermes.utils.offline.OfflineCardView;
+import com.isc.hermes.utils.offline.OfflineMapManager;
+import com.mapbox.mapboxsdk.geometry.LatLngBounds;
+import com.mapbox.mapboxsdk.offline.OfflineManager;
+import com.mapbox.mapboxsdk.offline.OfflineRegion;
+import com.mapbox.mapboxsdk.offline.OfflineRegionError;
+import com.mapbox.mapboxsdk.offline.OfflineRegionStatus;
+import com.mapbox.mapboxsdk.offline.OfflineTilePyramidRegionDefinition;
+
+import java.util.HashMap;
+import java.util.Map;
+import timber.log.Timber;
+
+/**
+ * This class represents the Offline Mode Settings UI.
+ */
+public class OfflineModeSettingsActivityView extends AppCompatActivity {
+
+    private LinearLayout vBoxDownloadedMaps;
+    private OfflineMapManager offlineMapManager;
+    public static Map<String, OfflineRegion> offlineRegionsNames;
+    private Map<String, CardView> cardViews;
+    private OfflineCardView offlineCardView;
+    private static final int REQUEST_CODE_OFFLINE = 231;
+
+    /**
+     * Method for creating the activity configuration.
+     *
+     * @param savedInstanceState If the activity is being re-initialized after
+     *                           previously being shut down, then this Bundle contains the data it most
+     *                           recently supplied in {@link #onSaveInstanceState}. Otherwise, it is null.
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.offline_mode_settings_view);
+        offlineMapManager = new OfflineMapManager(this);
+        offlineCardView = new OfflineCardView(this);
+        vBoxDownloadedMaps = findViewById(R.id.vBoxMapsDownloaded);
+    }
+
+
+
+    /**
+     * Navigates to the selected downloaded map.
+     */
+    protected void navigateToDownloadedMap(String regionName) {
+        OfflineRegion offlineRegion = offlineRegionsNames.get(regionName);
+        if (offlineRegion == null) return;
```

Comment:
Maybe is better log or raise an error if the `offlineRegion` cannot be found or why is it null.